packages/rich-text-links/src/index.ts

```diff
@@ -46,6 +48,29 @@ export function getRichTextResourceLinks(
   return iteratorToArray(links.values());
 }
 
+export function getAllRichTextResourceLinks(
```

Comment:
I can see a lot of logic being shared with `getRichTextResourceLinks`, actually just nodes you iterate over and `isValidType` are slightly different. Would it be possible to optimize this duplication?

Reply:
Why do we need a new function at all? 🤔 
I’d expect to get all links from `getRichTextResourceLinks`.

Reply:
@veu `getRichTextResourceLinks` gets one node type as a param. We can use it by calling it 3 times for `BLOCKS.EMBEDDED_RESOURCE`, `INLINES.EMBEDDED_RESOURCE` and `INLINES.RESOURCE_HYPERLINK`. Or we can refactor `getAllRichTextResourceLinks` to get an array as param. In that case it will be a breaking change. We have to update the usage wherever it's used. Is there any other option?

Reply:
hm another option would be to make that param in `getRichTextResourceLinks` optional, and with no filtering it gets all. That way you don't break existing usage, just enhances an existing fn

what I can see here is that there might be a need to allow users to pass specific node types (not just get all) for ex.

Reply:
@luizfonseca do you mean node types as an array?

Reply:
@alicandko yeah, something around those lines. But again, they can already do that by just calling the fn twice/thrice